### PR TITLE
PTI-257: Updating order of operations for add doc.

### PR DESCRIPTION
**Pull request by danieltruong**
Signed-off-by: Daniel Truong <daniel@dtsoftware.ca>

### CODE REVIEW

File: angular/projects/admin-nrpti/src/app/records/orders/order-add-edit/order-add-edit.component.ts [hunk 1]
```diff
@@ -263,13 +252,33 @@ export class OrderAddEditComponent implements OnInit, OnDestroy {
...
         this.router.navigate(['records', 'orders', this.currentRecord._id, 'detail']);
       });
     }
   }
 
+  private async handleDocumentChanges(recordId) {
+    const promises = [];
+
+    // Handle adding links
+    this.links.forEach(link => {
```
Comment (marklise) [inline angular/projects/admin-nrpti/src/app/records/orders/order-add-edit/order-add-edit.component.ts]: This should be async

File: angular/projects/admin-nrpti/src/app/records/orders/order-add-edit/order-add-edit.component.ts [hunk 2]
```diff
@@ -263,13 +252,33 @@ export class OrderAddEditComponent implements OnInit, OnDestroy {
...
+      promises.push(this.factoryService.createDocument(link, recordId));
+    });
+
+    // TODO: Handle adding S3 Docs
+
+    // TODO: Handle deleting documents
+
+    // Execute
+    const response = await Promise.all(promises);
```
Comment (marklise) [inline angular/projects/admin-nrpti/src/app/records/orders/order-add-edit/order-add-edit.component.ts]: No need to const then return it.  Just `return await Promise.all(promises);`

File: api/src/controllers/document-controller.js [hunk 3]
```diff
@@ -17,33 +19,48 @@ exports.protectedOptions = function(args, res, next) {
...
+    args.swagger.params.recordId &&
+    args.swagger.params.recordId.value
+  ) {
+    let data = args.swagger.params.data.value;
+    let docResponse;
+
+    if (data.url) {
+      // If the document already has a url we can assume it's a link
+      try {
+
```
Comment (marklise) [inline api/src/controllers/document-controller.js]: Eliminate this line.

File: angular/projects/admin-nrpti/src/app/records/orders/order-add-edit/order-add-edit.component.ts [hunk 4]
```diff
@@ -47,7 +47,7 @@ export class OrderAddEditComponent implements OnInit, OnDestroy {
     private factoryService: FactoryService,
     private utils: Utils,
     private _changeDetectionRef: ChangeDetectorRef
-  ) {}
+  ) { }
```
Comment (NickPhura) [inline angular/projects/admin-nrpti/src/app/records/orders/order-add-edit/order-add-edit.component.ts]: Something weird going on with your formatting settings.  It shouldn't be adding spaces between brackets like this.

File: angular/projects/admin-nrpti/src/app/services/document.service.ts [hunk 5]
```diff
@@ -13,7 +13,7 @@ import { HttpClient } from '@angular/common/http';
  */
 @Injectable({ providedIn: 'root' })
 export class DocumentService {
-  constructor(public apiService: ApiService, public http: HttpClient) {}
+  constructor(public apiService: ApiService, public http: HttpClient) { }
```
Comment (NickPhura) [inline angular/projects/admin-nrpti/src/app/services/document.service.ts]: Same as above, nothing here has really changed, except for a space between the brackets at the end.

**Pull request merged**